discord/ext/commands/core.py

```diff
@@ -1873,7 +1873,38 @@ def decorator(func):
 def before_invoke(coro):
     """A decorator that registers a coroutine as a pre-invoke hook.
 
-    Same as :func:`Command.before_invoke`.
+    This allows you to refer to one before invoke hook for several commands that
+    do not have to be within the same cog. 
+    
+    Example
+    ---------
+
+    .. code-block:: python3
+        
+        async def record_usage(ctx):
+            print(ctx.author, 'used', ctx.command, 'at', ctx.message.created_at)
+
+        @commands.before_invoke(record_usage)
```

Comment:
General style-guide for these tends to have the bot command decorator on top, not on bottom.